### DISCUSSION TITLE
Is there a way to retrieve the "revised_prompt" and used tokens when using the SemanticKernel Azure APIs

### DISCUSSION THREAD

**Discussion by efonsecab** [microsoft/semantic-kernel#6979] [Q&A] [answered]
Hello, I'm considering moving completely to SemanticKernel, and I'd like to know if currently there is a way to retrieve the revised_prompt when using the Azure OpenAI service through SemanticKernel

**Comment by sophialagerkranspandey** [comment]
Can you please confirm which API you're using? Is this for chat completion or image creation? Thanks!

**Reply by efonsecab** [reply]
> Can you please confirm which API you're using? Is this for chat completion or image creation? Thanks!

Image creation. Currently in my app I'm using this specific API, where the result object has the "revised_prompt" property.
https://platform.openai.com/docs/api-reference/images/create

From my tests using SemanticKernel, as far as I know, one has to use this interface: "ITextToImageService".

The problem I have is that the method "GenerateImageAsync" only returns a string which is the url of the image, so I have no idea how to get the "revised_prompt".

**Comment by sophialagerkranspandey** [comment, marked as answer]
We've created 2 issues from this. Thanks!
